### ISSUE TITLE
Two things to mention 

### ISSUE THREAD

**Issue by Crispy13** [IAmSuyogJadhav/3d-mri-brain-tumor-segmentation-using-autoencoder-regularization#29]
Hi. I'm using this source code to build a model for my project.

I found several things which are different from the paper or need to be fixed.

1. VAE branch is not trained.
    - When creating model using keras.models.Model, the output argument has not the VAE branch output.

2. Dropout was used instead of spatial dropout.
>We also use the spatial dropout with a rate of 0.2 after the initial encoder convolution.

3. When evaluating the model, dice score varys batch by batch.
Current dice score is not the mean of the scores(e.g. (sample 1 + sample 2) / 2, ) but the function calculates the score like the samples of a batch are one sample.

**Comment by IAmSuyogJadhav** [comment]
Hello @Crispy13, I am currently involved with a lot of other work and do not have spare time to focus on past projects.
I see you have created your own fork implementing fixes for some of the issues you faced. If you manage to fix any more and test them out for correctness, feel free to submit a PR here for other users to benefit from.
